packages/template-retail-react-app/app/components/_app/index.jsx

```diff
@@ -327,4 +343,50 @@ App.propTypes = {
     config: PropTypes.object
 }
 
+/**
+ * a hook that parallelly and individually fetches category based on the given ids
+ * @param ids - list of categories ids to fetch
+ * @param queryOptions -  react query options
+ * @return list of react query results
+ */
+export const useCategoryBulk = (ids, queryOptions) => {
+    if (!ids || ids.length === 0) {
+        return
+    }
+    const api = useCommerceApi()
+    const {getTokenWhenReady} = useAccessToken()
+    const {
+        app: {commerceAPI}
+    } = getConfig()
+    const {
+        parameters: {organizationId}
+    } = commerceAPI
+    const {site} = useMultiSite()
+
+    const queries = ids.map((id) => {
+        return {
+            queryKey: [
+                '/organizations/',
+                organizationId,
+                '/categories/',
+                id,
+                {id, levels: 2, organizationId, siteId: site.id}
+            ],
```

Comment:
```suggestion
            // Also needed:
            // import * as queryKeyHelpers from 'commerce-sdk-react-preview/hooks/ShopperProducts/queryKeyHelpers'
            queryKey: queryKeyHelpers.getCategory.queryKey({id, levels: 2, organizationId, siteId: site.id}),
```
1. That's not the query key any more, it's prefixed with `/commerce-sdk-react` now!
2. If we use the query key helpers, we never need to care what the query key actually is.